packages/plugin-essentials/sources/commands/add.ts

```diff
@@ -133,19 +133,23 @@ export default class AddCommand extends BaseCommand {
 
     const fixed = this.fixed;
     const interactive = this.interactive ?? configuration.get(`preferInteractive`);
+    const reuse = interactive || configuration.get(`preferReuse`);
 
     const modifier = suggestUtils.getModifier(this, project);
 
     const strategies = [
-      ...interactive ? [
-        suggestUtils.Strategy.REUSE,
-      ] : [],
+      reuse ?
+        suggestUtils.Strategy.REUSE
+        : undefined,
+
       suggestUtils.Strategy.PROJECT,
-      ...this.cached ? [
-        suggestUtils.Strategy.CACHE,
-      ] : [],
+
+      this.cached ?
+        suggestUtils.Strategy.CACHE
+        : undefined,
+
       suggestUtils.Strategy.LATEST,
-    ];
+    ].filter(isStrategy);
```

Comment:
```suggestion
    ].filter((strategy): strategy is suggestUtils.Strategy => typeof strategy !== `undefined`);
```
I would prefer to have this inlined here